### PR TITLE
[FLINK-28655][SQL gateway] Support show jobs operation in SqlGateway

### PR DESCRIPTION
**Pull request by link3280**
## What is the purpose of the change

Support show jobs statement in SqlGatewayService. Subtask of FLIP-222.

## Brief change log

- Add show jobs statement in SQL parser.
- Add show jobs operation support in the table planner.
- Add show jobs operation support in SQL gateway.

## Verifying this change

This change added tests and can be verified as follows:

- Extend FlinkSqlParserImplTest to validate the parsing of show-jobs statements.
- Extend SqlGatewayServiceITCase to validate the execution of show-jobs statements.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by link3280** [comment]
@flinkbot run azure

**Comment by link3280** [comment]
@flinkbot run azure

**Comment by link3280** [comment]
Please kindly take a look @fsk119 .

**Review by LadyForest — COMMENTED** [review]
Hi @link3280, thanks for the contribution. I hope you don't mind an unsolicited review.
I left some questions that are not clear to me from the FLIP doc.

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationExecutor.java [hunk 1]
```diff
@@ -524,6 +533,47 @@ private ResultFetcher buildOkResultFetcher(OperationHandle handle) {
...
+                                    return clusterClient.listJobs().get();
+                                } catch (Exception e) {
+                                    throw new FlinkException(e);
+                                }
+                            });
+            List<RowData> resultRows =
+                    jobs.stream()
+                            .map(
+                                    job ->
+                                            GenericRowData.of(
```
[the remark this answers is not in the document]
Comment (link3280) [reply flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationExecutor.java:554]: It requires changes to `JobStatusMessage` and `RestClient` of `flink-client` module, so I thought it'd be clear to leave the rest columns to another PR. But I'm OK to add them in a PR if you think that's better.

File: flink-table/flink-sql-parser/src/main/codegen/data/Parser.tdd [hunk 2]
```diff
@@ -144,6 +145,7 @@
     "JAR"
     "JARS"
     "JOB"
+    "JOBS"
```
[the remark this answers is not in the document]
Comment (link3280) [reply flink-table/flink-sql-parser/src/main/codegen/data/Parser.tdd:148]: You're right.

**Comment by link3280** [comment]
> I hope you don't mind an unsolicited review.
No, absolutely not! Thanks a lot for your review! @LadyForest

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/utils/Constants.java [hunk 3]
```diff
@@ -22,9 +22,12 @@
 public class Constants {
 
     public static final String JOB_ID = "job id";
+    public static final String JOB_NAME = "job name";
+    public static final String STATUS = "status";
+    public static final String START_TIME = "start time";
     public static final String SET_KEY = "key";
     public static final String SET_VALUE = "value";
     public static final String COMPLETION_HINTS = "hints";
```
Comment (fsk119) [inline flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/utils/Constants.java:31]: nit: I think we can remove this empty line

**Review by fsk119 — COMMENTED** [review]
Thasnks for your update. I left some comments.

**Comment by link3280** [comment]
@fsk119 Thanks a lot for your review! The PR is updated as suggested.

**Review by fsk119 — APPROVED** [review]
LGTM

**Comment by LadyForest** [comment]
Thanks for the update; looks good to me too.

**Comment by link3280** [comment]
Thanks a lot! @LadyForest @fsk119

**Pull request merged**